src/Common/Bridge/BridgeEntityFacadeFetchFunction.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace ArchitectureSniffer\Common\Bridge;
+
+use ArchitectureSniffer\Zed\Dependency\Bridge\AbstractBridgeRule;
+use PHPMD\AbstractNode;
+use PHPMD\Node\MethodNode;
+use PHPMD\Rule\MethodAware;
+
+class BridgeEntityFacadeFetchFunction extends AbstractBridgeRule implements MethodAware
+{
+    /**
+     * @return string
+     */
+    public function getDescription(): string
+    {
+        return 'An entity fetching function in a facade should always follow the following rules:
+        - Name follows convention: getDomainEntityCollection
+        - Have a transfer object based on Domain Entity name (e.g. DomainEntityCriteriaTransfer) as input
+        - Return a transfer object with a collection containing results of fetching';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function apply(AbstractNode $node): void
+    {
+        if (!$this->isBridge($node) || !$this->isEntityFetchingFacadeBridge($node)) {
+            return;
+        }
+
+        $this->applyRule($node);
+    }
+
+    /**
+     * @param \PHPMD\AbstractNode $node
+     *
+     * @return bool
+     */
+    protected function isEntityFetchingFacadeBridge(AbstractNode $node): bool
+    {
+        $namespaceParts = explode('\\', $node->getNode()->getParent()->getNamespace()->getName());
+
+        if ($namespaceParts && count($namespaceParts) > 2 && in_array($namespaceParts[(count($namespaceParts) - 1)], ['Client', 'Facade'])) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param \PHPMD\Node\MethodNode $method
+     *
+     * @return void
+     */
+    protected function applyRule(MethodNode $method): void
+    {
+        if (!preg_match('/(get|find|fetch).+/', $method->getName())) {
```

Comment:
```suggestion
        if (!preg_match('/^(get|find|fetch).+/', $method->getName())) {
```
You need to be careful with false positives on regexp matching